pkg/sql/logictest/testdata/logic_test/privilege_builtins

```diff
@@ -1054,3 +1054,76 @@ query B
 SELECT has_column_privilege('hcp_test'::REGCLASS, 4, 'SELECT')
 ----
 true
+
+# Regression test for #58254. Tests privilege inheritance for roles (direct and indirect).
+
+user root
+
+statement ok
+DROP DATABASE IF EXISTS my_db;
+CREATE DATABASE my_db;
+
+statement ok
+CREATE ROLE my_role;
+
+statement ok
+GRANT CREATE ON DATABASE my_db TO my_role;
+GRANT my_role TO testuser;
+
+user testuser
+
+statement ok
+use my_db
+
+statement ok
+CREATE SCHEMA s;
+CREATE TABLE s.t()
+
+# Privilege check on direct member of role with CREATE privilege granted.
+query B
+SELECT has_schema_privilege('testuser', 's', 'create')
+----
+true
+
+# Privilege check on direct member of role without USAGE privilege granted.
+query B
+SELECT has_schema_privilege('testuser', 's', 'usage')
+----
+false
+
+# Confirm privilege checks on all objects.
+query BBB
+SELECT has_database_privilege('testuser', 'my_db', 'create'),
+       has_schema_privilege('testuser', 'public', 'usage'),
+       has_table_privilege('testuser', 's.t', 'select')
+----
+true false false
+
+user root
+
+statement ok
+CREATE USER testuser2;
+GRANT testuser TO testuser2;
+GRANT ALL ON SCHEMA s TO testuser2
```

Comment:
Seems to be failing here 

```logic.go:2357:
testdata/logic_test/privilege_builtins:1104:
expected success, but found
(3F000) unknown schema "s"
errors.go:60: in NewUndefinedSchemaError()
logic.go:2693:
testdata/logic_test/privilege_builtins:1104: error while processing
logic.go:2693: pq: unknown schema "s"
```

Does it need to be `my_db.s`? I'm not sure why it fails in 20.2 but not in the original PR.

Reply:
ah i see... i had to remove that because this isn't (and won't be) backported to 20.2: https://github.com/cockroachdb/cockroach/pull/55647

hmm but i would have expected it to work since the current db is my_db.w will investigate

Reply:
ah i see, probably because the user was switched to root.